### ISSUE TITLE
UI: admin cannot take volume snapshot on user VM

### ISSUE THREAD

**Issue by levindecaro** [apache/cloudstack#6795]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.17.1
~~~

##### CONFIGURATION

kvm
SharedMountPoint (CephFS)

##### OS / ENVIRONMENT

CentOS Linux release 7.9.2009 (Core) 3.10.0-1160.45.1.el7.x86_64
libvirt-4.5.0-36.el7_9.5.x86_64
qemu-kvm-ev-2.12.0-44.1.el7_8.1.x86_64
qemu-img-ev-2.12.0-44.1.el7_8.1.x86_64
qemu-kvm-common-ev-2.12.0-44.1.el7_8.1.x86_64
libvirt-daemon-driver-qemu-4.5.0-36.el7_9.5.x86_64

##### STEPS TO REPRODUCE

~~~
1. Login as admin
2. navigate to compute -> instances
3. select one of user's VM
4. click "Take VM volume snapshot"
5. no volume display for selection.
~~~

##### EXPECTED RESULTS

~~~
Allow admin take volume snapshot on user's VM
~~~

##### ACTUAL RESULTS

~~~
No volume shown in volume snapshot on user's VM

~~~

![image](https://user-images.githubusercontent.com/36956864/193400057-2eedd98c-00db-4a56-9ce2-3221bcca7f32.png)

**Comment by DaanHoogland** [comment]
@levindecaro , does it work with the API, i.e. is this a UI bug only?

**Comment by levindecaro** [comment]
API appear ok, likely UI bug.

(?) 🐱 > create snapshot volumeid=030b678b-96b2-4e56-b4c9-4c434e282abf asyncbackup=true
{
  "snapshot": {
    "account": "levin.ng",
    "created": "2022-10-04T17:28:22+0800",
    "domain": "varmeego",
    "domainid": "ab262bfb-f415-41c7-98a9-196a3f4da096",
    "hasannotations": false,
    "id": "f93c0ca4-be81-46a9-9f44-58dd6ed364e1",
    "intervaltype": "MANUAL",
    "name": "centos9_ROOT-4170_20221004092822",
    "osdisplayname": "CentOS 9",
    "ostypeid": "6efc8758-2863-11ec-97da-52540044e551",
    "physicalsize": 0,
    "revertable": true,
    "snapshottype": "MANUAL",
    "state": "BackingUp",
    "tags": [],
    "virtualsize": 10737418240,
    "volumeid": "030b678b-96b2-4e56-b4c9-4c434e282abf",
    "volumename": "ROOT-4170",
    "volumetype": "ROOT",
    "zoneid": "5e43ac5b-d657-49f2-98e1-7304476d28ed"
  }
}
(?) 🐱 >

**Comment by kohrar** [comment]
The problem seems to be that the UI code doesn't include `listall=true` in the `listVolumes` API call so it's not getting any volumes. When I click on 'take VM volume snapshot', this is the HTTP request that gets generated:

```
GET /client/api/?virtualMachineId=a1691e94-14ec-4207-a9cf-8689c5f18a4f&command=listVolumes&response=json HTTP/1.1
```

Running the same command via `cmk` shows the same behavior, where the result is empty unless `listall=true` is included.
```
(localcloud) 🐱 > list volumes virtualmachineid=a1691e94-14ec-4207-a9cf-8689c5f18a4f filter=id,size,state
(localcloud) 🐱 >

(localcloud) 🐱 > list volumes virtualmachineid=a1691e94-14ec-4207-a9cf-8689c5f18a4f listall=true filter=id,size,state
count = 3
volume:
+--------------------------------------+---------------+-------+
|                  ID                  |     SIZE      | STATE |
+--------------------------------------+---------------+-------+
| ff6014c3-6689-4b15-b651-d62068e804b0 | 2199023255552 | Ready |
| 0b9f07e5-5758-4a60-9059-cc0c2c76a9bf |  536870912000 | Ready |
| 6f835800-9b44-49ae-a6e6-abb744295a87 |   32212254720 | Ready |
+--------------------------------------+---------------+-------+

```

**Comment by kohrar** [comment]
@nvazquez Thanks for the PR. I've tested it and the volumes now show up in the dropdown as expected.

**Comment by DaanHoogland** [comment]
fixed by #6824
